src/pages/workspace/members/WorkspaceOwnerChangeWrapperPage.tsx

```diff
@@ -0,0 +1,100 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useEffect, useState} from 'react';
+import {View} from 'react-native';
+import type {ValueOf} from 'type-fest';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import withPolicy from '@pages/workspace/withPolicy';
+import type {WithPolicyOnyxProps} from '@pages/workspace/withPolicy';
+import * as PolicyActions from '@userActions/Policy';
+import CONST from '@src/CONST';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import WorkspaceOwnerChangeCheck from './WorkspaceOwnerChangeCheck';
+import WorkspaceOwnerPaymentCardForm from './WorkspaceOwnerPaymentCardForm';
+
+type WorkspaceOwnerChangeWrapperPageProps = WithPolicyOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.OWNER_CHANGE_CHECK>;
+
+function WorkspaceOwnerChangeWrapperPage({route, policy}: WorkspaceOwnerChangeWrapperPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const [isTransitioning, setIsTransitioning] = useState(false);
+
+    const policyID = route.params.policyID;
+    const accountID = route.params.accountID;
+    const error = route.params.error;
+
+    useEffect(() => {
+        setIsTransitioning(true);
+    }, []);
+
+    useEffect(() => {
+        if (!policy || policy?.isLoading) {
+            return;
+        }
+
+        if (!policy.errorFields && policy.isChangeOwnerFailed) {
+            // there are some errors but not related to change owner flow - show an error page
+            Navigation.navigate(ROUTES.WORKSPACE_OWNER_CHANGE_ERROR.getRoute(policyID, accountID));
+            return;
+        }
+
+        if (!policy?.errorFields?.changeOwner && policy?.isChangeOwnerSuccessful) {
+            // no errors - show a success page
+            Navigation.navigate(ROUTES.WORKSPACE_OWNER_CHANGE_SUCCESS.getRoute(policyID, accountID));
+            return;
+        }
+
+        const changeOwnerErrors = Object.keys(policy?.errorFields?.changeOwner ?? {});
+
+        if (changeOwnerErrors && changeOwnerErrors.length > 0 && changeOwnerErrors[0] !== CONST.POLICY.OWNERSHIP_ERRORS.NO_BILLING_CARD) {
+            Navigation.navigate(ROUTES.WORKSPACE_OWNER_CHANGE_CHECK.getRoute(policyID, accountID, changeOwnerErrors[0] as ValueOf<typeof CONST.POLICY.OWNERSHIP_ERRORS>));
```

Comment:
coming from to this issue https://github.com/Expensify/App/issues/57245, when we encounter an `changeOwnerErrors` value, we navigate to `WORKSPACE_OWNER_CHANGE_CHECK`. However, this leads to the same page being added to the navigation stack since we are already on `WORKSPACE_OWNER_CHANGE_CHECK`. As a result, pressing the back button on the success transfer page takes us to a blank page because the same screen exists twice with different parameters and to reach the member details page, we then have to press back twice. Instead, we can simply update the params of the current page if we receive a different error value from the backend. More context can be found here: https://github.com/Expensify/App/issues/57245#issuecomment-2676151355